Implement generalized gamma distribution

kylejcaron opened the pull request:
**Thank you for opening a PR!**

This PR implements an Op for a Generalized Gamma Random Variable with the parameterization GG(alpha, p, lambd). This corresponds to scipy in the following mapping: `scipy.stats.gengamma(a=alpha/p, c=p, scale=lambd)`.

This goes along with the following [PyMC PR to implement the Generalized Gamma distribution](https://github.com/pymc-devs/pymc/pull/5985)

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [x] There is an informative high-level description of the changes.
+ [ ] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [x] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [ ] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.
+ [x] There are tests covering the changes introduced in the PR.

Don't worry, your PR doesn't need to be in perfect order to submit it.  As development progresses and/or reviewers request changes, you can always [rewrite the history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) of your feature/PR branches.

If your PR is an ongoing effort and you would like to involve us in the process, simply make it a [draft PR](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

brandonwillard reviewed (commented):
Looks good, although the commits need to be squashed/rebased before merging.

kylejcaron commented:
> Looks good, although the commits need to be squashed/rebased before merging (because there's a merge commit in this branch).

just rebased and squashed! Any advice on how to coordinate this PR with the corresponding PyMC PR?

brandonwillard commented:
> Any advice on how to coordinate this PR with the corresponding PyMC PR?

You can increment the required Aesara version in your PyMC PR; that will at least guarantee that your changes will only apply to a version of Aesara with these changes.
